app/coffee/Features/Compile/ClsiManager.coffee

```diff
@@ -101,56 +117,116 @@ module.exports = ClsiManager =
 		return outputFiles
 
 	VALID_COMPILERS: ["pdflatex", "latex", "xelatex", "lualatex"]
+
 	_buildRequest: (project_id, options={}, callback = (error, request) ->) ->
-		Project.findById project_id, {compiler: 1, rootDoc_id: 1, imageName: 1}, (error, project) ->
+		ProjectGetter.getProject project_id, {compiler: 1, rootDoc_id: 1, imageName: 1, rootFolder:1}, (error, project) ->
 			return callback(error) if error?
 			return callback(new Errors.NotFoundError("project does not exist: #{project_id}")) if !project?
-
 			if project.compiler not in ClsiManager.VALID_COMPILERS
 				project.compiler = "pdflatex"
 
+			if options.incremental or options.syncType? # new way, either incremental or full
```

Comment:
this `options.incremental` is a feature flag I think? It reads as if it will make it an `incremental` compile at the moment, so perhaps call this `options.incrementalCompilesEnabled`?